databases/iplike/Portfile

```diff
@@ -14,20 +15,19 @@ long_description    C implementation of OpenNMS's plpgsql iplike function, \
                     preferred as it offers better performance.  Allows for \
                     complex IP address queries on a text field.
 
-homepage            http://www.opennms.org/wiki/IPLIKE
+depends_lib         port:postgresql16
 
-master_sites        sourceforge:opennms
-
-depends_lib         port:postgresql90
-
-depends_run         port:postgresql90-server
-
-checksums           rmd160  167d520c82824b1b3c6cc4ea0799bafa4cab70c4 \
-                    sha256  52703c5186ab2f8c5de86214a40ebfde3eb29454f7695ed84fa5b5d257c09eef \
-                    size    358568
-
-configure.args      --with-pgsql=${prefix}/lib/postgresql90/bin/pg_config
+fetch.type          git
+post-fetch {
+    system -W ${worksrcpath} "git submodule update --init"
+}
```

Comment:
do we really need this, can you not just depend on the `m4` port instead?

Reply:
Upstream's sort of a mess. They're on an ancient autotools. The m4 directory here isn't for m4 itself but some custom autotools functionality implemented in m4. Their recommended build directions are to do this submodule checkout because the GitHub releases do not pull it in and they haven't uploaded a autoreconf-ed tarball.

Reply:
I think we have a strong prejudice against doing network based fetches during build if we can possibly avoid it.

Reply:
(I have never used the feature, but we can do multiple separate file fetches during fetch.)

Reply:
I copied this verbatim from the macports guide https://guide.macports.org/#reference.portgroup.github.submodule

Reply:
>"If the project uses git submodules, some projects' tag- or commit-based distfiles will not contain all the necessary files. Once again, the best distfile candidate (if available) is a distfile from GitHub releases, as described in the previous sections. However, in the case a project doesn't provide any other alternative, a project using submodules can be successfully retrieved by fetching the sources using git and then using a post-fetch to initialize the submodules:"

This does not say "use git submodules preferentially". This says "use them if there is no way to get a distfile. In the current case, a distfile could be fetched, right?

Reply:
No, none of the tarballs from GitHub include the m4 directory's contents.

Reply:
But that can in fact be fetched by pointing at the upstream repo.

Reply:
well, there are many ports who do the "git submodule" stuff. So while indeed not ideal, I think we can live with it...